### PR TITLE
Whisper v0.1

### PR DESCRIPTION
**Pull request by chandralegend**
- You can parse an Audiofile, URL, Audio Array and get a transcription.
- Support 99 languages
- Support Translation (Given french audio can get a English translation)

### CODE REVIEW

**Comment by ypkang** [comment]
@chandralegend 
* I am having trouble running this on my machine. I tried it as a server and via running the unit tests and get the following error.
```
Traceback (most recent call last):
  File "/home/ypkang/.local/lib/python3.10/site-packages/librosa/core/audio.py", line 164, in load
    y, sr_native = __soundfile_load(path, offset, duration, dtype)
  File "/home/ypkang/.local/lib/python3.10/site-packages/librosa/core/audio.py", line 195, in __soundfile_load
    context = sf.SoundFile(path)
  File "/home/ypkang/.local/lib/python3.10/site-packages/soundfile.py", line 655, in __init__
    self._file = self._open(file, mode_int, closefd)
  File "/home/ypkang/.local/lib/python3.10/site-packages/soundfile.py", line 1213, in _open
    raise LibsndfileError(err, prefix="Error opening {0!r}: ".format(self.name))
soundfile.LibsndfileError: Error opening '726c19bd-3c63-4b90-96dc-a39b150d0a7c.wav': File contains data in an unknown format.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/ypkang/.local/lib/python3.10/site-packages/jaseci_ai_kit/modules/whisper/whisper.py", line 197, in translate
    audio_array = get_array(downloaded_audio_file)
  File "/home/ypkang/.local/lib/python3.10/site-packages/jaseci_ai_kit/modules/whisper/whisper.py", line 127, in get_array
    audio, _ = librosa.load(audio_file, sr=16000)
  File "/home/ypkang/.local/lib/python3.10/site-packages/librosa/util/decorators.py", line 88, in inner_f
    return f(*args, **kwargs)
  File "/home/ypkang/.local/lib/python3.10/site-packages/librosa/core/audio.py", line 170, in load
    y, sr_native = __audioread_load(path, offset, duration, dtype)
  File "/home/ypkang/.local/lib/python3.10/site-packages/librosa/core/audio.py", line 226, in __audioread_load
    reader = audioread.audio_open(path)
  File "/home/ypkang/.local/lib/python3.10/site-packages/audioread/__init__.py", line 132, in audio_open
    raise NoBackendError()
audioread.exceptions.NoBackendError
```
* Let's not call the module whisper. Let's call it `speech_to_text`, to be consistent with the rest of the module where the module name explains what the module does but not the underlying model used.

File: README.md [hunk 1]
```diff
@@ -87,6 +87,7 @@
   - [USE Encoder](jaseci_ai_kit/jaseci_ai_kit/modules/use_enc/README.md)
   - [USE QA](jaseci_ai_kit/jaseci_ai_kit/modules/use_qa/README.md)
   - [Personalized Head](jaseci_ai_kit/jaseci_ai_kit/modules/ph/README.md)
+  - [Whisper](jaseci_ai_kit/jaseci_ai_kit/modules/whisper/README.md)
```
Comment (ypkang) [inline README.md]: Can you also create a new entry in the readme at the top of jaseci_ai_kit? You can create a new table for speech related modules.

File: jaseci_ai_kit/jaseci_ai_kit/modules/whisper/README.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
+# **Whisper (`whisper`)**
```
Comment (ypkang) [inline jaseci_ai_kit/jaseci_ai_kit/modules/whisper/README.md]: This readme is great! 💯

File: jaseci_ai_kit/jaseci_ai_kit/modules/whisper/requirements.txt [hunk 3]
```diff
@@ -0,0 +1,3 @@
+transformers>=4.25.1
+librosa>=0.9.2
+soundfile>=0.11.0
```
Comment (ypkang) [inline jaseci_ai_kit/jaseci_ai_kit/modules/whisper/requirements.txt]: I don't see soundfile in the setup.py. Do we need to add it?

**Review by ypkang — CHANGES_REQUESTED** [review]
See comment a

File: README.md (hunk 1 above)
Comment (ypkang) [reply README.md]: Can you also make one for the translation module too? I just merged that one.

**Comment by chandralegend** [comment]
> @chandralegend
> 
> * I am having trouble running this on my machine. I tried it as a server and via running the unit tests and get the following error.
> 
> ```
> Traceback (most recent call last):
>   File "/home/ypkang/.local/lib/python3.10/site-packages/librosa/core/audio.py", line 164, in load
>     y, sr_native = __soundfile_load(path, offset, duration, dtype)
>   File "/home/ypkang/.local/lib/python3.10/site-packages/librosa/core/audio.py", line 195, in __soundfile_load
>     context = sf.SoundFile(path)
>   File "/home/ypkang/.local/lib/python3.10/site-packages/soundfile.py", line 655, in __init__
>     self._file = self._open(file, mode_int, closefd)
>   File "/home/ypkang/.local/lib/python3.10/site-packages/soundfile.py", line 1213, in _open
>     raise LibsndfileError(err, prefix="Error opening {0!r}: ".format(self.name))
> soundfile.LibsndfileError: Error opening '726c19bd-3c63-4b90-96dc-a39b150d0a7c.wav': File contains data in an unknown format.
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "/home/ypkang/.local/lib/python3.10/site-packages/jaseci_ai_kit/modules/whisper/whisper.py", line 197, in translate
>     audio_array = get_array(downloaded_audio_file)
>   File "/home/ypkang/.local/lib/python3.10/site-packages/jaseci_ai_kit/modules/whisper/whisper.py", line 127, in get_array
>     audio, _ = librosa.load(audio_file, sr=16000)
>   File "/home/ypkang/.local/lib/python3.10/site-packages/librosa/util/decorators.py", line 88, in inner_f
>     return f(*args, **kwargs)
>   File "/home/ypkang/.local/lib/python3.10/site-packages/librosa/core/audio.py", line 170, in load
>     y, sr_native = __audioread_load(path, offset, duration, dtype)
>   File "/home/ypkang/.local/lib/python3.10/site-packages/librosa/core/audio.py", line 226, in __audioread_load
>     reader = audioread.audio_open(path)
>   File "/home/ypkang/.local/lib/python3.10/site-packages/audioread/__init__.py", line 132, in audio_open
>     raise NoBackendError()
> audioread.exceptions.NoBackendError
> ```
> 
> * Let's not call the module whisper. Let's call it `speech_to_text`, to be consistent with the rest of the module where the module name explains what the module does but not the underlying model used.

Cause of error is the ffmpeg. I will add a line on README on how to install ffmpeg.

File: jaseci_ai_kit/jaseci_ai_kit/modules/whisper/requirements.txt (hunk 3 above)
Comment (chandralegend) [reply jaseci_ai_kit/jaseci_ai_kit/modules/whisper/requirements.txt]: SOundfile get installed with librosa

**Review by ypkang — APPROVED** [review]

**Pull request merged**